### PR TITLE
fix(apple/java): Improve docs for attachments

### PR DESCRIPTION
**Pull request by philipphofmann**
The docs for attachments didn't mention content type. Now the docs are
restructured so they give more insight on how to use attachments.

Fixes GH-2824

### CODE REVIEW

File: src/platforms/common/enriching-events/attachments/index.mdx [hunk 1]
```diff
@@ -41,6 +41,31 @@ To receive symbolicated stack traces, you have to upload debug information to Se
...
+<PlatformContent includePath="attachment-init-with-path" />
+
+Another way to initialize an attachment is to pass the `bytes`. When doing so you also need to specify a filename.
+
+<PlatformContent includePath="attachment-init-with-bytes" />
+
+If your SDK supports offline caching, which is typical for mobile, each attachment is stored to disk for each event or transaction you capture. When using large
+attachments this can eat up the disk space if the device is offline for a longer time period. You can use [maximum attachment size](#maximum-attachment-size) to drop large attachments.
+
+Additional parameters:
```
Comment (PeloWriter) [inline src/platforms/common/enriching-events/attachments/index.mdx:61]:
```suggestion
In addition, you can set these parameters:
```

File: src/platforms/common/enriching-events/attachments/index.mdx [hunk 2]
```diff
@@ -41,6 +41,31 @@ To receive symbolicated stack traces, you have to upload debug information to Se
...
+
+**`filename`**: The filename is the name of the file to display in Sentry. When using bytes you have to specify a filename, whereas with a path you don't as the SKD is going to use the last path component.
+
+**`contentType`**: The `contentType` of the attachment. The default is `application/octet-stream`.
```
Comment (PeloWriter) [inline src/platforms/common/enriching-events/attachments/index.mdx:65]:
```suggestion
`filename`

: The filename is the name of the file to display in Sentry. When using bytes you have to specify a filename, whereas with a path you don't as the SKD is going to use the last path component.

`contentType`

: The type of content stored in this attachment. The default is `application/octet-stream`.
```

**Review by PeloWriter — APPROVED** [review]
Nicely done. A few edits to fix format for `filename` and `contentType` and smooth the prose.

**Comment by borninla** [comment]
Not sure if it belongs in this specific article, but it would be nice to talk about some (if not most) of the options that can be specified for `contentType`.

**Comment by philipphofmann** [comment]
Thanks a lot @PeloWriter for the feedback 👏🏼

**Comment by philipphofmann** [comment]
@borninla all media types specified here: https://www.iana.org/assignments/media-types/media-types.xhtml. I'm going to add this to the docs.

**Comment by philipphofmann** [comment]
I opened up two PRs @borninla https://github.com/getsentry/develop/pull/238 and https://github.com/getsentry/sentry-docs/pull/2843.

**Pull request merged**